pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternMultiTopicsConsumerImpl.java

```diff
@@ -99,13 +113,48 @@ public static NamespaceName getNameSpaceFromPattern(Pattern pattern) {
         return TopicName.get(pattern.pattern()).getNamespaceObject();
     }
 
+    /**
+     * This method will be called after the {@link TopicListWatcher} reconnected after enabled {@link TopicListWatcher}.
+     */
+    private void recheckTopicsChangeAfterReconnect() {
+        // Skip if closed or the task has been cancelled.
+        if (getState() == State.Closing || getState() == State.Closed) {
+            return;
+        }
+        // Do check.
+        recheckTopicsChange().whenComplete((ignore, ex) -> {
```

Comment:
Will the `recheckTopicsChange` method be called in two threads concurrently?

Reply:
Yes, it will

Reply:
Should we make it sequentially?

Reply:
Connection opend and back retry could call this method, I think it's ok to call 2 times in this case.

Reply:
@Demogorgon314 

> Should we make it sequentially?

It will be sequentially because multi tasks will use the same `connection`, and then they will be executed one by one.

Reply:
OK, thanks for explaining.

Reply:
@Demogorgon314 

But if users set `maxConnectionsPerHosts` more significant than `1`, it will break the thread safety; I will fix it.

Reply:
@Demogorgon314 @Technoboy- 

Fixed. Please take a look again.

Reply:
LGTM.